### PR TITLE
Responsiveness

### PR DESCRIPTION
**Pull request by aspitz1**
What does this PR do?
* Adds a breakpoint for small screen layouts
* Refactors some CSS to be hopefully sturdier
* Includes button that leads to form from prescription
* The refactoring of the logo to be a link made it real weird on a small screen
* Header probably needs to be refactored to be friendlier to small layouts

Where should the reviewer start?
* This is a big PR, pull it down to test responsiveness

Anything else the reviewer needs to know (e.g. issues, concerns)?
* Still needs a medium screen breakpoint 
* It May be nice to pull down the code and follow along with comments, there was a lot to refactor to make the responsiveness work

TasteRX (3)

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -4,26 +4,11 @@
...
     <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
-    <!--
-      Notice the use of %PUBLIC_URL% in the tags above.
-      It will be replaced with the URL of the `public` folder during the build.
-      Only files inside the `public` folder can be referenced from the HTML.
-
-      Unlike "/favicon.ico" or "favicon.ico", "%PUBLIC_URL%/favicon.ico" will
-      work correctly both with client-side routing and a non-root public URL.
-      Learn how to configure a non-root public URL by running `npm run build`.
-    -->
```
Comment (aspitz1) [inline public/index.html:26]: Cleaned up this file to remove unneeded code

File: public/manifest.json [hunk 2]
```diff
@@ -6,16 +6,6 @@
...
-    },
-    {
-      "src": "logo192.png",
-      "type": "image/png",
-      "sizes": "192x192"
-    },
-    {
-      "src": "logo512.png",
-      "type": "image/png",
-      "sizes": "512x512"
```
Comment (aspitz1) [inline public/manifest.json:18]: Deleted both these logos as they where unused

File: src/components/App/App.css [hunk 3]
```diff
@@ -4,7 +4,6 @@
   text-align: center;
   justify-content: flex-end;
   align-items: center;
-  height: 100%;
```
Comment (aspitz1) [inline src/components/App/App.css:7]: This height isn't necessary, it is helpful not to have it when components will overflow

File: src/components/App/App.css [hunk 4]
```diff
@@ -19,7 +18,7 @@
   display: flex;
   flex-direction: column;
   align-items: center;
-  height: 90%;
+  height: 70vh;
```
Comment (aspitz1) [inline src/components/App/App.css:21]: Making this view height helped with a bit of styling, I honestly can not recall where.

File: src/components/App/App.css [hunk 5]
```diff
@@ -46,9 +45,32 @@
   background-color: white;
 }
 
-.prescription-page {
-  display: flex;
-  justify-content: center;
-  height: 80%;
-  width: 45%;
-}
```
Comment (aspitz1) [inline src/components/App/App.css:55]: I removed this and instead styled the prescription size, vs having it fill the parent. This just took away a layer of complexity.

File: src/components/App/App.css [hunk 6]
```diff
@@ -46,9 +45,32 @@
...
+    width: 100%;
+  }
+
+  .prescription-container {
+    width: 100%;
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+  }
+}
```
Comment (aspitz1) [inline src/components/App/App.css]: Adds style for screens smaller than 576px, which is a common phone size.

File: src/components/App/App.js [hunk 7]
```diff
@@ -59,26 +59,26 @@ function App() {
     <div className="App">
       <Routes>
         <Route exact path='/' element={
-          <div className='home-page'>
+          <main className='home-page'>
```
Comment (aspitz1) [inline src/components/App/App.js:62]: Changed to main to make it more symantic.

File: src/components/App/App.js [hunk 8]
```diff
@@ -59,26 +59,26 @@ function App() {
...
             <Header />
-            <div className='search-form-container'>
+            <section className='search-form-container'>
```
Comment (aspitz1) [inline src/components/App/App.js:64]: Changed to section to make it more symantic

File: src/components/Header/Header.css [hunk 9]
```diff
@@ -14,4 +14,23 @@
...
+
+  .title-logo {
+    position: static;
+    height: 100%;
+  }
+
+  .logo {
+    position: static;
+    height: 100%;
+  }
```
Comment (aspitz1) [inline src/components/Header/Header.css:40]: The header needs some refactoring. The way it is styled using absolute positioning isn't great for responsive layouts. There isn't enough room on a phone screen to make something like this work well. I tried to set it to a phone-friendly layout the best. It isn't great and needs to be reworked.

File: src/components/Instructions/Instructions.css [hunk 10]
```diff
@@ -1,9 +1,11 @@
 .instructions {
-  display: flex;
-  flex-direction: column;
-  align-items: center;
-  justify-content: space-evenly;
-  height: 100%;
-  width: 90%;
-  font-size: 1vw;
```
Comment (aspitz1) [inline src/components/Instructions/Instructions.css:8]: Generally speaking, percentages and vw/vh aren't great for styling fonts. They don't need to grow and shrink in the same way as some other elements. I Changed it where I saw it. I made a lot of fonts smaller for the time being.

File: src/components/Prescription/Prescription.css [hunk 11]
```diff
@@ -1,12 +1,21 @@
...
+    display: flex;
+    flex-direction: column;
+    align-items: flex-start;
+    margin-top: 22vh;
     border: black solid 3px;
     background-color: white;
-    height: 88%;
-    width: 95%; 
+    height: 70vh;
+    width: 40vw;
```
Comment (aspitz1) [inline src/components/Prescription/Prescription.css:9]: Here is where I refactored this to rely on its own size vs the container it is in.

File: src/components/Prescription/Prescription.css [hunk 12]
```diff
@@ -1,12 +1,21 @@
...
-    width: 95%; 
+    height: 70vh;
+    width: 40vw; 
     border-radius: 10px;
+}
+
+.top-container {
     display: flex;
-    align-items: flex-start;
-    flex-wrap: wrap;
```
Comment (aspitz1) [inline src/components/Prescription/Prescription.css:9]: This seemed to rely on the wrap to control the layout, when I removed it the layout broke. I refactored the component to not rely on this. I think it is a sturdier layout.

File: src/components/Prescription/Prescription.css [hunk 13]
```diff
@@ -1,12 +1,21 @@
...
+    flex-direction: row;
+    justify-content: space-between;
+    width: 100%;
+    height: 50%;
 }
```
Comment (aspitz1) [inline src/components/Prescription/Prescription.css:19]: The containers in this component got moved around to facilitate this layout, just an fyi.

File: src/components/Prescription/Prescription.css [hunk 14]
```diff
@@ -15,7 +24,18 @@
...
+ }
+
+ .logo-show-details-container {
+    padding-left: 5%;
+ }
+
+.script-show-poster {
+    height: 90%;
+    margin: 3%;
+    aspect-ratio: auto;
```
Comment (aspitz1) [inline src/components/Prescription/Prescription.css:36]: This aspect-ratio keeps the image from distorting.

File: src/components/ShowDetails/ShowDetails.css [hunk 15]
```diff
@@ -17,34 +19,42 @@
 }
 
 .prompt {
-  margin: 20% 0% 0% 15%;
+  white-space: nowrap;
```
Comment (aspitz1) [inline src/components/ShowDetails/ShowDetails.css:22]: This font was wrapping where it didn't need to.

File: src/components/ShowDetails/ShowDetails.css [hunk 16]
```diff
@@ -17,34 +19,42 @@
...
 }
 
 .rx-logo {
   height: 60%;
-  width: 20%;
+  aspect-ratio: auto;
```
Comment (aspitz1) [inline src/components/ShowDetails/ShowDetails.css:27]: Keeps the image from distorting.

File: src/components/ShowDetails/ShowDetails.css [hunk 17]
```diff
@@ -17,34 +19,42 @@
...
-  font-size: 135%;
-
-}
-
-.poster-container {
-  display: flex;
-  justify-content: center;
-  align-items: center;
-  height: 100%;
-  width: 40%;
```
Comment (aspitz1) [inline src/components/ShowDetails/ShowDetails.css:42]: This didn't seem to affect anything when I removed it.

File: src/components/ShowDetails/ShowDetails.css [hunk 18]
```diff
@@ -17,34 +19,42 @@
...
+    height: 70%;
+    margin: 15% 10px;
+  }
+
+  .show-name {
+    font-size: 1em;
+  }
+
+  .prompt {
+    white-space: normal;
```
Comment (aspitz1) [inline src/components/ShowDetails/ShowDetails.css:57]: Needed to wrap here.

File: src/index.css [hunk 19]
```diff
@@ -14,6 +14,7 @@ code {
 
 html {
   background-image: linear-gradient(#6CD0E7, #9DEA8D);
+  background-attachment: fixed;
```
Comment (aspitz1) [inline src/index.css:17]: Keeps the background from repeating when there is an overflow.

Comment (dstavis) [reply src/index.css:17]: Was it repeating somewhere? I don't remember seeing it repeat.

**Review by tfettig22 — APPROVED** [review]

**Pull request merged**